images/scripts/services.setup

```diff
@@ -71,8 +66,13 @@ poweroff
 #
 #############
 
-# See https://hub.docker.com/r/nowsci/samba-domain
-docker_pull nowsci samba-domain
+# See https://github.com/Fmstrat/samba-domain
+# image on dockerhub is too old, and not published on quay or anywhere else
+git clone --depth=1 https://github.com/Fmstrat/samba-domain /tmp/samba-domain
```

Comment:
I know that this is basically what we were doing before with pulling the latest version of whatever container image happened to be there, but for some reason, with git, this looks a lot worse.  Probably because the Dockerfile can also do stuff on the "host"?  (even though the "host" is, itself, just a VM)...

Just wanted to ask if you considered (and dismissed) the idea to pin a specific SHA here...

Reply:
`podman build` also happens inside of a container. You mean that a Dockerfile can somehow break out of that?

I didn't consider a specific SHA, as with our test images we want the "latest" stuff for everything, and there is nothing interesting/secret in our test images. In general we could of course start from scratch from a Fedora or Ubuntu container, and inject some stuff -- but (1) this is not actually trivial, and (2) it's pretty much exactly what this repo does. In essence, it's "someone else already did that work".

Reply:
> You mean that a Dockerfile can somehow break out of that?

If that was possible, how does dockerhub, quay, other services which build containers for you not hopelessly explode?

Reply:
so a Dockerfile, for example, can easily copy files from the host.  I'm not well-versed in all of the features of what you can do from there, but it's clear that there is at least some interaction with the host environment...

Reply:
That way around, yes -- after all, the Dockerfile itself comes from the host. I'm just not aware of a container build being able to change anything on the host, and the VM host has no secrets.

Reply:
OK.  WMF.